src/UI/Component/Modal/Factory.php

```diff
@@ -0,0 +1,161 @@
+	 *     overriding workflow. This is typically the case if an ILIAS service is called
+	 *     for while in working in an object.
+	 *   composition: >
+	 *     Round-Trip Modals are completed by a well defined sequence of only a few steps that might be
+	 *     displayed on a sequence of different modals connected through some "next" Button.
+	 *   effect: >
+	 *     Round-Trip Modals perform sub-workflow involving some kind of user input. Sub-workflow is completed
+	 *     and user is returned to starting point allowing for continuing the overriding workflow.
+	 *
+	 * rules:
+	 *   usage:
+	 *     1: >
+	 *       Round-Trip Modals MUST contain at least two buttons at the bottom of the modals: a button
+	 *       to cancel (right) the workflow and a button to finish or reach the next step in the workflow (left).
+	 *     2: >
+	 *       Round-Trip Modals SHOULD be used, if the user would lose the context otherwise. If the action
+	 *       can be performed within the same context (e.g. add a post in a forum, edit a wiki page),
+	 *       a Round-Trip Modal MUST NOT be used.
+	 *     3: >
+	 *       When the workflow is completed, Round-Trip Modals SHOULD show the same view as
+	 *       was displayed when initiating the Modal.
+	 *     4: >
+	 *       Round-Trip Modals SHOULD NOT be used to add new items of any kind since adding item is a
+	 *       linear workflow redirecting to the newly added item setting- or content-tab.
+	 *     5: >
+	 *       Round-Trip Modals SHOULD NOT be used to perform complex workflows.
+	 *
+	 * ---
+	 * @param string                                    $title
+	 * @param Component\Component|Component\Component[] $content
+	 *
+	 * @return \ILIAS\UI\Component\Modal\RoundTrip
+	 */
+	public function roundtrip($title, $content);
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     The Lightbox Modal displays media data such as documents, images or a movie inside a Lightbox.
+	 *   composition: >
+	 *     If multiple pages or images are to be displayed in one Lightbox Modal, they can flipped through.
+	 *   effect: >
+	 *      Lightbox Modals are activated by clicking the Full View Glyphicon,
+	 *      the Title of the object or it's Thumbnail.
+	 *
+	 * rules:
+	 *   usage:
+	 *     1: >
+	 *       Lightobx Modals MUST contain a title above the presented item.
+	 *     2: >
+	 *       Lightbox Modals SHOULD contain a descriptional text below the presented items.
+	 *     3: >
+	 *       Multiple images or pages inside a Lightbox Modals MUST be presented in carousel
+	 *       like manner allowing to flickr through items.
+	 *
+	 * ---
+	 * @param LightboxPage|LightboxPage[] $pages
+	 *
+	 * @return \ILIAS\UI\Component\Modal\Lightbox
+	 */
+	public function lightbox($pages);
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     Used to display an image component inside a lightbox modal
+	 *   composition:
+	 *   effect:
```

Comment:
Please describe effects of those pages (e.g. what happens if images are changed).